core/src/test/java/org/testcontainers/utility/DockerImageNameCompatibilityTest.java

```diff
@@ -0,0 +1,94 @@
+package org.testcontainers.utility;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertFalse;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
+
+
+public class DockerImageNameCompatibilityTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testPlainImage() {
+        DockerImageName subject = DockerImageName.parse("foo");
+
+        assertFalse("image name foo != bar", subject.isCompatibleWith(DockerImageName.parse("bar")));
+    }
+    @Test
+    public void testLatestTreatedAsWildcard() {
+        final DockerImageName subject = DockerImageName.parse("foo:4.5.6");
+
+        assertTrue("foo:4.5.6 ~= foo:latest", subject.isCompatibleWith(DockerImageName.parse("foo:1.2.3").withTag("latest")));
+    }
+
+    @Test
+    public void testImageWithAutomaticCompatibility() {
```

Comment:
after my suggestion, this would be the same test as `testLatestTreatedAsWildcard`

Reply:
Removed